ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/jts/RecoveryManagerService.java

```diff
@@ -54,11 +54,18 @@ public RecoveryManagerService(org.omg.CORBA.ORB theCorbaORB) throws Exception
 
         org.omg.PortableServer.POA rootPOA = org.omg.PortableServer.POAHelper.narrow(theCorbaORB.resolve_initial_references("RootPOA"));
 
-        orb.setOrb(theCorbaORB);
+        orb.setOrb(new ORBWrapper(theCorbaORB));
         OA oa = OA.getRootOA(orb);
         oa.setPOA(rootPOA);
 
         ORBManager.setORB(orb);
         ORBManager.setPOA(oa);
     }
+
+    @Override
+    public void stop() throws Exception {
+        super.stop();
+
+        ORB.getInstance("jboss-atx").shutdown();
```

Comment:
It must be the same one as was constructed in the start method so I guess we could do with a final field for that.